tests/e2e/e2evalidator/e2evalidator.go

```diff
@@ -6,11 +6,30 @@ import (
 	"strings"
 
 	"github.com/go-playground/validator/v10"
+	corev1 "k8s.io/api/core/v1"
 )
 
+type Ruler interface {
+	Cmp(val any) error
+}
+
 type Rule string
 
-func StructWithRules(actual any, rulesMapping map[string]Rule) (errMessages []string) {
+func (r Rule) Cmp(val any) error {
+	return validator.New().Var(val, string(r))
+}
+
+type Resource corev1.ResourceRequirements
```

Comment:
Perhaps the definition could be more abstract rather than specific to a certain type.

Reply:
ok